File: Snappet.Contracts/Assesments/ViewModels/ClassModel.cs
using Snappet.Contracts.Assesments.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace Snappet.Contracts.Assesments.ViewModels
{
    public class ClassModel
    {
        public int Id { get; set; }        
        public string Period { get; set; }
        public List<StatiscticsModel> TotalProgress { get; set; }        
        public List<SubjectModel> ResultsPerSubjects { get; set; }
        public DateTime CurrentDate { get; set; }
        public string PreviousResultType { get; set; }

        public ClassModel()
        {
            TotalProgress = new List<StatiscticsModel>();
            ResultsPerSubjects = new List<SubjectModel>();
        }
    }
}

File: SnappetSolution/ClientApp/app/models/classModel.ts
import { SubjectModel } from "./subjectModel";
import { StudentModel } from "./StudentModel";

export class ClassModel {
    public Id: number = 0;
    public Period: string = "";
    public CurrentDate: Date;
    public PreviousResultType: string;
    public Subjects: Array<string> = [];
    public StudentsModel: Array<StudentModel> = [];
}